src/main/scala/hellotest/Main.scala

```diff
@@ -62,10 +62,10 @@ object Main:
     }
 
     // 4. discard sequences of non-valid length
-    val validLengthSequences = stringSequences // DO:
+    val validLengthSequences = stringSequences.filter(seq => seq.length >= args.minLength)
```

Comment:
You can simplify this a bit, so instead of using `filter`, use `drop` to discard the first sequences that have less than the required number of elements based on `args.windowSize`.
Ex:
```
stringSequences.drop(args.windowSize - 1)
```
This should work since you we know that the first sequences will be like this:
```
["aa"]
["aa","bb"]
["aa","bb","cc"]
["aa","bb","cc","dd"]
["aa","bb","cc","dd","ee"]
```
Assuming `args.windowSize` equals `5`, we'll want to discard the first four sequences

Reply:
Hey Adam I tried implementing the valid sequence logic using drop but I ran into too many issues with skipping sequences that were in fact valid, for now all test cases but I'll look into this more and see if I can have it implemented this way, for now it does seem to be working.

Reply:
I think I made a mistake with the logic, but I think what you did is right. I'm just trying to figure out something about the final result of the run function, so I'll send another message in a bit